linera-storage-service/src/client.rs

```diff
@@ -54,14 +57,18 @@ const MAX_KEY_SIZE: usize = 1000000;
 //   [KeyTag::Key] + [namespace] + [key]
 // * An additional key with empty value is stored at
 //   [KeyTag::Namespace] + [namespace]
-// is stored to indicate the existence of a namespace.
+//   is stored to indicate the existence of a namespace.
+// * A key with empty value is stored at
+//   [KeyTag::RootKey] + [namespace] + [root_key]
+//   to indicate the existence of a root key.
 #[derive(Clone)]
 pub struct ServiceStoreClientInternal {
     channel: Channel,
     semaphore: Option<Arc<Semaphore>>,
     max_stream_queries: usize,
-    namespace: Vec<u8>,
+    prefix_len: usize,
     start_key: Vec<u8>,
+    root_key_written: Arc<Mutex<bool>>,
```

Comment:
Maybe an `AtomicBool` would be lower overhead. Also, you could do `.fetch_or(true)` below to set and get the value atomically.

Reply:
Yes, thanks.

Reply:
It's still a mutex in this file.

Reply:
I was going to ask why :)